File: solutions/Count-and-Say/Count-and-Say.cpp

class Solution {
public:
    string countAndSay(int n) {
        string ans="1";
        for(int i=1;i<n;i++)
        {
            stringstream ss;
            int cnt=0;
            char last=ans[0];
            for(int j=0;j<=ans.length();j++)
            {
                if(ans[j]==last)
                    cnt++;
                else
                {
                    ss<<cnt<<last;
                    cnt=1;
                    last=ans[j];
                }
            }
            ans=ss.str();
        }
        return ans;
    }
};


 

File: solutions/Count-and-Say/Count-and-Say.js
/**
 * @param {number} n
 * @return {string}
 */
var countAndSay = function (n) {
  let ans = '1';
  for (let i = 1; i < n; i++) {
    const length = ans.length;
    let cnt = 0;
    let last = ans[0];
    let temp = '';
    for (let j = 0; j <= length; j++) {
      if (ans[j] === last) {
        cnt++;
      } else {
        temp += cnt + last;
        cnt = 1;
        last = ans[j];
      }
    }
    ans = temp;
  }
  return ans;
};